src/binding/petsc4py/src/petsc4py/PETSc/Scatter.pyx

```diff
@@ -48,72 +130,199 @@ cdef class Scatter(Object):
+
+        See Also
+        --------
+        setType, petsc.VecScatterGetType
+
+        """
         cdef PetscScatterType cval = NULL
         CHKERR( VecScatterGetType(self.sct, &cval) )
         return bytes2str(cval)
 
-    def setFromOptions(self):
+    def setFromOptions(self) -> None:
+        """Configure the scatter from the options database.
+
+        Collective.
+
+        See Also
+        --------
+        petsc_options, petsc.VecScatterSetFromOptions
+
+        """
         CHKERR( VecScatterSetFromOptions(self.sct) )
 
-    def setUp(self):
+    def setUp(self) -> None:
+        """Set up the internal data structures for using the scatter.
+
+        Collective.
+
+        See Also
+        --------
+        petsc.VecScatterSetUp
+
+        """
         CHKERR( VecScatterSetUp(self.sct) )
         return self
 
-    def copy(self):
+    def copy(self) -> Scatter:
+        """Return a copy of the scatter"""
         cdef Scatter scatter = Scatter()
         CHKERR( VecScatterCopy(self.sct, &scatter.sct) )
         return scatter
 
     @classmethod
-    def toAll(cls, Vec vec):
+    def toAll(cls, Vec vec) -> Scatter:
+        """Create a scatter that communicates a vector to all sharing processes.
+
+        Collective.
+
+        Parameters
+        ----------
+        vec
+          The vector to scatter from. The resulting scatter will have the same communicator.
+
+        See Also
+        --------
+        toZero, petsc.VecScatterCreateToAll
+
+        """
         cdef Scatter scatter = Scatter()
         cdef Vec ovec = Vec()
         CHKERR( VecScatterCreateToAll(
             vec.vec, &scatter.sct, &ovec.vec) )
         return (scatter, ovec)
 
     @classmethod
-    def toZero(cls, Vec vec):
+    def toZero(cls, Vec vec) -> Scatter:
+        """Create a scatter that communicates a vector to rank zero of the sharing communicator.
+
+        Collective.
+
+        Parameters
+        ----------
+        vec
+          The vector to scatter from. The resulting scatter will have the same communicator.
+
+        See Also
+        --------
+        toAll, petsc.VecScatterCreateToZero
+
+        """
         cdef Scatter scatter = Scatter()
         cdef Vec ovec = Vec()
         CHKERR( VecScatterCreateToZero(
             vec.vec, &scatter.sct, &ovec.vec) )
         return (scatter, ovec)
     #
 
-    def begin(self, Vec vec_from, Vec vec_to, addv=None, mode=None):
-        cdef PetscInsertMode  caddv = insertmode(addv)
-        cdef PetscScatterMode csctm = scattermode(mode)
-        CHKERR( VecScatterBegin(self.sct, vec_from.vec, vec_to.vec,
-                                caddv, csctm) )
-
-    def end(self, Vec vec_from, Vec vec_to, addv=None, mode=None):
-        cdef PetscInsertMode  caddv = insertmode(addv)
-        cdef PetscScatterMode csctm = scattermode(mode)
-        CHKERR( VecScatterEnd(self.sct, vec_from.vec, vec_to.vec,
-                              caddv, csctm) )
+    scatterBegin = begin
```

Comment:
You have to define `scatterBegin/End` AFTER implementing the methods `begin/end`. Maybe it works in Cython, but just because of implementation details.

Reply:
done